### DISCUSSION TITLE
Thread full end device

### DISCUSSION THREAD

**Discussion by eerke2021** [seaky/nspanel_pro_tools_apk#59] [Ideas]
it is possible to change the built in radio to be a Zigbee Router. I believe the Zigbee radio can also run thread firmware, which is available from Silicon Labs. 

Do you think it is possible to install Thread Full End Device firmware? Obviously not Thread Border Router functionality as that'd be more than just the Radio Hardware. Ditto combined Zigbee/Thread functionality. Thread Full End Device though is similar to a Zigbee Router. 

I was tempted to see if I could access the radio chip directly and flash it but thought I'd see if you thought it was possible in software? 

(I already have a strong Zigbee network so making my thread network stronger would be nice)

**Comment by seaky** [comment]
Changing built in radio to Zigbee router is quite easy you can find a thread here:
https://github.com/seaky/nspanel_pro_roottool_apk/issues/3

I'll integrate it to v2.2 if you are sharped-eyed  you may have noticed it in the manual there will be an integration tab in v2.1 where I prepared the place for Zigbee integration

**Reply by eerke2021** [reply]
No, I must not have been clear enough. The radio chip, if it is what I think it is, can be either a Zigbee radio or a Thread radio (or both with software control). 
The idea is just to add Silicon Labs firmware for thread in place of Zigbee. 

For example you can put alternative firmware on the Sonoff Dongle E

https://smarthomescene.com/guides/how-to-enable-thread-and-matter-support-on-sonoff-zbdongle-e/

The Thread only firmware is probably on the Silicon Labs website but that is easy to find.
